metadata_retriever.go

```diff
@@ -0,0 +1,40 @@
+package lifecycle
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/platform"
+)
+
+type CacheMetadataRetriever interface {
```

Comment:
Should we move the interface to the package where it's used?

Reply:
I put them here because I didn't want to increase the surface area of the API we export in the `lifecycle` or `buildpacks` or `platform`. But I am fine with putting it in one of those buckets. Are you thinking `platform`?

Reply:
Oh hmm... I guess it's `main` that uses these. I would prefer not to put them in `platform` because that package feels like it should be very implementation agnostic and mostly reflect spec'd behavior.

Reply:
We could also put these into a new `internal/` package, if you rather.

Reply:
I think it's a good idea to create an `internal` package but we'll probably want to be a bit more deliberate about how we go about it. I'm fine leaving the interface where it is right now.